src/accelerate/scheduler.py

```diff
@@ -61,7 +61,9 @@ def step(self, *args, **kwargs):
             # num_processes steps per training step
             num_processes = AcceleratorState().num_processes
             for _ in range(num_processes):
-                self.scheduler.step(*args, **kwargs)
+                # Special case when using OneCycle and `drop_last` was not used
+                if getattr(self.scheduler, "total_steps", 0) <= self.scheduler.last_epoch:
```

Comment:
Should we guard around the `last_epoch` attribute here as well somehow?

Reply:
`last_epoch` is part of the base scheduler object, so it's always present: https://github.com/pytorch/pytorch/blob/master/torch/optim/lr_scheduler.py#L24